### ISSUE TITLE
Why need yes_loc?

### ISSUE THREAD

**Issue by lhwcv** [FlagOpen/FlagEmbedding#1163]
Thank you for your great work first.

If there any  special consideration?  
and what's the advantage 

```
    def encode(self, features):
        # input('continue?')
        if features is None:
            return None
        outputs = self.model(input_ids=features['input_ids'],
                             attention_mask=features['attention_mask'],
                             position_ids=features['position_ids'] if 'position_ids' in features.keys() else None,
                             output_hidden_states=True)
        _, max_indices = torch.max(features['labels'], dim=1)
        predict_indices = max_indices - 1
        logits = [outputs.logits[i, predict_indices[i], :] for i in range(outputs.logits.shape[0])]
        logits = torch.stack(logits, dim=0)
        scores = logits[:, self.yes_loc]
        return scores.contiguous()
```

**Comment by 545999961** [comment]
Similar to [monoT5](https://arxiv.org/abs/2003.06713) and [RankT5](https://arxiv.org/abs/2210.10634), we expect to use the logits of yes as the score.

**Comment by lhwcv** [comment]
Thank you
